### ISSUE TITLE
File drag and drop bar of Extension 'Ms-Upload' 

### ISSUE THREAD

**Issue by nauman5** [Mediawiki-wysiwyg/WYSIWYG-CKeditor#99]
Extension MS-Upload allow users to upload files via drag and drop bar (visible just below WikiEditor toolbar). However when I use WYSIWYG extension, I do not see this drag and drop bar.

I am using MediaWiki 1.26.2 with IE 11

I'd appreciate your help in resolving this issue.

**Comment by glodoski** [comment]
I don't believe the issue is due to the WYSIWYG-CKeditor plugin here.  

The MSUpload drag-and-drop plugin works just fine in our environment (the drag-and-drop shows up above the entirety of the CKeditor window).

Here is our relevant Special:Version page output:

MediaWiki: 1.26.2
MSUpload: 12.0
WYSIWYG-CKeditor: Wysiwyg 1.5.6_0 [B551+09.11.2016], CKEditor 4.5.9 (revision a35abfe)
WikiEditor: 0.4.0 
(the version included in WYSIWYG-CKeditor source, and the LocalSettings.php line uncommented: require_once( "$IP/extensions/WikiEditor/WikiEditor.php" );  

Edit:
I don't really know why we have to have WikiEditor enabled, but without it, the WYSIWYG editor doesn't even show up at all, and as it works just fine I never felt like digging into why we couldn't disable WikiEditor altogether.

**Comment by riparap** [comment]
Thank you for your comments.
Reason why you have to have WikiEditor enabled with latest versions of Wysiwyg is json definition file for resourceloader of MW. I do not know how conditional loading of javascript files can be accomplished. The way javascripts are configured is not "by the book", but it seems to be good enough in order to make Wysiwyg start with latest version of MW. F.ex usage of source files for debug purposes requires some manual tweaks.

I have tried MSupload plugin once but could not get it work with Wysiwyg, unfortunateldy I do not remember anymore what versions I was using at that time. Does it work with IE, FF and Chrome too in your environment?

**Comment by glodoski** [comment]
It appears to work just fine.  Unfortunately our Wiki contains sensitive, internal data and we can only access it on our internal network, so you can't see it in action, but I've attached a screen shot of the edit page for an article.  The "Drop Files Here" area has been skinned with CSS, but it is still there and fully functional.

![screen shot 2016-11-22 at 1 39 31 pm](https://cloud.githubusercontent.com/assets/10697847/20539115/914300ce-b0b9-11e6-8553-14a840c11961.png)

**Comment by riparap** [comment]
That looks really great :).  Seems like you have editor section, upload area at top and categories at the bottom. Would it be possible for you to provide instructions how you have set it up like that either here or in MediaWiki page of Wysiwyg?

**Comment by nauman5** [comment]
Yes, it would be good to know how editing section has been set like this specially the category selection drop down. 
I am using MSUpload version 10.3 with WYSIWYG version 1.5.6_0 [B551+08.07.2016] and CKEditor 4.5.8 rev c1fc9a9 but drag and drop doesn't show up. Also drag and drop area in my case appears below the WikiEditor (if I stop using WYSIWYG) and not above.

Not sure why my MSUpload version is 10.3 and yours is 12 even though I downloaded it a week ago.

**Comment by glodoski** [comment]
Sorry for the delay - the Thanksgiving holiday week was short but crammed.

The category selection is simply the [MsCatSelect](https://www.mediawiki.org/wiki/Extension:MsCatSelect) extension.

The WYSIWYG editor has the [MoonoColor](http://ckeditor.com/addon/moonocolor) theme downloaded and then the contents of the plugin's kama theme folder replaced with the contents of the MoonoColor theme.

The general formatting comes from the [Chameleon](https://www.mediawiki.org/wiki/Skin:Chameleon) skin in FixedHead mode.

The Common.css file has a number of customizations for spacing and colors.

According to the [MsUpload](https://www.mediawiki.org/wiki/Extension:MsUpload) page, the latest version is indeed 12.0. 

I install my plugins via wget and my LocalSettings.php is edited by nano within an SSH session on Ubuntu 14.04 LTS.

**Comment by nauman5** [comment]
Hi James, Thank you for outlining your configuration details.

**Comment by riparap** [comment]
Thank you for details. Would it be possible for you to copy-paste those css here ?
